### ISSUE TITLE
Segfault when switching between drumkits of different size and immediately quiting the application

### ISSUE THREAD

**Issue by theGreatWhiteShark** [hydrogen-music/hydrogen#1474]
Hydrogen version *          : all
Operating system + version  : all 
Audio driver + version      :  all

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Steps to reproduce:
1. New song in Pattern Mode with `GMRockKit` loaded and all notes on the *Bell* instrument (bottom most one) filled
2. Load `TR808EmulationKit` and hit keep
3. Load `GMRockKit`
4. Load `TR808EmulationKit` and hit discard
5. Quit Hydrogen (immediately)

I'm not quite sure why this does not show up when quiting after step 2. But maybe I'm just to slow with the mouse.

I suspect the `Hydrogen::__kill_instruments()` or something related to be the cause.

**Comment by theGreatWhiteShark** [comment]
Better way to reproduce:

1. as above
2. store the song created in 1.
3. run this `bash` script

``` bash
hydrogen -s "/path/to/saved/song.h2song" &

sleep 5

oscsend localhost 9000 /Hydrogen/PLAY
sleep 2
oscsend localhost 9000 /Hydrogen/LOAD_DRUMKIT sf "TR808EmulationKit" 1
sleep 1
oscsend localhost 9000 /Hydrogen/LOAD_DRUMKIT sf "GMRockKit" 0
sleep 1
oscsend localhost 9000 /Hydrogen/LOAD_DRUMKIT sf "TR808EmulationKit" 0
sleep 5
oscsend localhost 9000 /Hydrogen/QUIT
```

When having multiple patterns + song mode and the bell is not active in all of them I do not get a segfault but a number of objects being still alive after quitting.

```
Objects map :                        class	constr   destr   alive
	[            AudioEngineInfoForm ]	     1	     1	     0
	[               InstrumentEditor ]	     1	     1	     0
	[                       Director ]	     1	     1	     0
	[             FilesystemInfoForm ]	     1	     1	     0
	[                    HydrogenApp ]	     1	     1	     0
	[          InstrumentEditorPanel ]	     1	     1	     0
	[                   LayerPreview ]	     1	     1	     0
	[                    WaveDisplay ]	     2	     2	     0
	[                 InstrumentRack ]	     1	     1	     0
	[             LadspaFXProperties ]	     4	     4	     0
	[                       MainForm ]	     1	     1	     0
	[                          Mixer ]	     1	     1	     0
	[           InstrumentNameWidget ]	    19	    19	     0
	[                      MixerLine ]	    18	    18	     0
	[             ComponentMixerLine ]	     1	     1	     0
	[                MasterMixerLine ]	     1	     1	     0
	[              LadspaFXMixerLine ]	     4	     4	     0
	[                  PatternEditor ]	     7	     7	     0
	[    PatternEditorInstrumentList ]	     1	     1	     0
	[                PianoRollEditor ]	     1	     1	     0
	[                       LCDCombo ]	     4	     4	     0
	[                     LCDDisplay ]	    32	    32	     0
	[             PatternEditorPanel ]	     1	     1	     0
	[             PatternEditorRuler ]	     1	     1	     0
	[                  PlayerControl ]	     1	     1	     0
	[                 PlaylistDialog ]	     1	     1	     0
	[                     LCDSpinBox ]	     9	     9	     0
	[                     SongEditor ]	     1	     1	     0
	[          SongEditorPatternList ]	     1	     1	     0
	[        SongEditorPositionRuler ]	     1	     1	     0
	[                         Button ]	   178	   178	     0
	[                SongEditorPanel ]	     1	     1	     0
	[              SoundLibraryPanel ]	     1	     1	     0
	[               SoundLibraryTree ]	     1	     1	     0
	[                   SplashScreen ]	     1	     1	     0
	[             AutomationPathView ]	     1	     1	     0
	[                 ClickableLabel ]	    50	    50	     0
	[                  CpuLoadWidget ]	     1	     1	     0
	[                            LED ]	    40	    40	     0
	[                   MetronomeLED ]	     1	     1	     0
	[                  CommonStrings ]	     1	     1	     0
	[                     Instrument ]	  1714	  1713	     1
	[                         Action ]	  1550	  1550	     0
	[                        XMLNode ]	 12332	 12332	     0
	[                         XMLDoc ]	    60	    60	     0
	[                          Theme ]	     1	     1	     0
	[           SoundLibraryDatabase ]	     1	     1	     0
	[               SoundLibraryInfo ]	     8	     8	     0
	[                          Fader ]	    21	    21	     0
	[                   PixmapWidget ]	    72	    72	     0
	[                         Rotary ]	   111	   111	     0
	[                    AudioEngine ]	     1	     1	     0
	[                  TransportInfo ]	     1	     1	     0
	[       AutomationPathSerializer ]	     1	     1	     0
	[                           ADSR ]	  3528	  3523	     5
	[                 AutomationPath ]	     1	     1	     0
	[                        Drumkit ]	    60	    60	     0
	[               DrumkitComponent ]	    65	    65	     0
	[            InstrumentComponent ]	  1771	  1770	     1
	[                InstrumentLayer ]	  5393	  5388	     5
	[                 InstrumentList ]	   121	   121	     0
	[                           Note ]	    71	    67	     4
	[                        Pattern ]	    10	    10	     0
	[                    PatternList ]	     8	     8	     0
	[                       Playlist ]	     1	     1	     0
	[                         Sample ]	  5392	  5387	     5
	[                     SongReader ]	     1	     1	     0
	[                           Song ]	     1	     1	     0
	[           CoreActionController ]	     1	     1	     0
	[                     EventQueue ]	     1	     1	     0
	[                        Effects ]	     1	     1	     0
	[                   LadspaFXInfo ]	   508	   508	     0
	[                       Hydrogen ]	     1	     1	     0
	[                    AudioOutput ]	     1	     1	     0
	[                AlsaAudioDriver ]	     1	     1	     0
	[                      MidiInput ]	     1	     1	     0
	[                     MidiOutput ]	     1	     1	     0
	[                 AlsaMidiDriver ]	     1	     1	     0
	[                   LocalFileMng ]	     1	     1	     0
	[              MidiActionManager ]	     1	     1	     0
	[                        MidiMap ]	     1	     1	     0
	[                      NsmClient ]	     1	     1	     0
	[                      OscServer ]	     1	     1	     0
	[               WindowProperties ]	    71	    71	     0
	[                    Preferences ]	     1	     1	     0
	[                     ColorTheme ]	     1	     1	     0
	[                 InterfaceTheme ]	     1	     1	     0
	[                      FontTheme ]	     1	     1	     0
	[                        Sampler ]	     1	     1	     0
	[                          Synth ]	     1	     1	     0
	[                       Timeline ]	     3	     3	     0
Total :     21 objects.
```

**Comment by theGreatWhiteShark** [comment]
Closed with #1496
